### PR TITLE
Allows better changing of movement. (with events and stuff)

### PR DESCRIPTION
**Pull request by PixelTheKermit**
## About the PR

Made friction and acceleration changeable with events.
Also allows the adjustment of weightless movement speeds.

## Why / Balance

Need this for downstream.

## Technical details

If you want to update acceleration, call RefreshMovementAccelerationModifiers.
If you want to update friction, call RefreshMovementFrictionModifiers.

You should also hook onto RefreshMovementAccelerationModifiersEvent and RefreshMovementFrictionModifiersEvent in your system if you want to adjust them.

## Media

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

### CODE REVIEW

File: Content.Shared/Movement/Systems/MovementSpeedModifierSystem.cs [hunk 1]
```diff
@@ -50,6 +50,28 @@ public void ChangeFriction(EntityUid uid, float friction, float? frictionNoInput
...
+            if (!Resolve(uid, ref move, false))
+                return;
+
+            move.WeightlessModifier = weightlessModifier ?? move.WeightlessModifier;
+            move.WeightlessAcceleration = weightlessAcceleration ?? move.WeightlessAcceleration;
+            move.WeightlessFriction = weightlessFriction ?? move.WeightlessFriction;
+            move.WeightlessFrictionNoInput = weightlessFrictionNoInput ?? move.WeightlessFrictionNoInput;
+            Dirty(uid, move);
+        }
     }
```
Comment (metalgearsloth) [inline Content.Shared/Movement/Systems/MovementSpeedModifierSystem.cs:75]: Please just make it a setter for 1 field, not 4.

Comment (PixelTheKermit) [reply Content.Shared/Movement/Systems/MovementSpeedModifierSystem.cs:75]: Sorry but could you please explain? My brain is lacking the thinking juices right now.

Comment (PixelTheKermit) [reply Content.Shared/Movement/Systems/MovementSpeedModifierSystem.cs:75]: Wait I think I figured it out never mind.

File: Content.Shared/Movement/Components/MovementSpeedModifierComponent.cs [hunk 2]
```diff
@@ -63,26 +63,26 @@ private float _baseSprintSpeedVV
         /// The negative velocity applied for friction when weightless and providing inputs.
         /// </summary>
         [ViewVariables(VVAccess.ReadWrite), DataField]
-        public float WeightlessFriction = DefaultWeightlessFriction;
+        public float WeightlessFriction { get; set; } = DefaultWeightlessFriction;
```
Comment (EmoGarbage404) [inline Content.Shared/Movement/Components/MovementSpeedModifierComponent.cs:66]: sloth meant make one function per variable, not adding the auto properties to the fields.

**Review by EmoGarbage404 — CHANGES_REQUESTED** [review]

File: Content.Shared/Movement/Components/MovementSpeedModifierComponent.cs (hunk 2 above)
Comment (PixelTheKermit) [reply Content.Shared/Movement/Components/MovementSpeedModifierComponent.cs]: actually just realized the comp doesn't get dirtied this way nevermind my question

File: Content.Shared/Movement/Systems/MovementSpeedModifierSystem.cs (hunk 1 above)
Comment (metalgearsloth) [reply Content.Shared/Movement/Systems/MovementSpeedModifierSystem.cs]: Also why isn't this just using the event?

**Pull request closed without merging**